### PR TITLE
[FLINK-27620][sql] Flink SQL supports PERCENT_RANK function

### PR DESCRIPTION
**Pull request by luoyuxia**
## What is the purpose of the change
To support PERCENT_RANK function

## Brief change log
- Add implementation for [PERCENT_RANK](https://github.com/apache/flink/pull/20347/files#diff-50dfaea3dd3c4981b6d098388215860d0142172bc32c1930ad73e37715647642)

## Verifying this change
[UT](https://github.com/apache/flink/pull/20347/files#diff-81346deaed0bf56b2b392761a0be36cb3d25ace2c6999b7a1c22f797e7c07615R2812) for PERCENT_RANK

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala [hunk 1]
```diff
@@ -105,6 +105,13 @@ class AggFunctionFactory(
           throw new TableException("CUME_DIST Function is not supported in stream mode.")
         }
 
+      case a: SqlRankFunction if a.getKind == SqlKind.PERCENT_RANK =>
+        if (isBounded) {
+          createPercentRankAggFunction(argTypes)
+        } else {
+          throw new TableException("PERCENT_RANK Function is not supported in batch mode.")
```
Comment (nyingping) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala:112]: `is not supported in stream mode` ?

Comment (luoyuxia) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala:112]: Yes,  `PERCENT_RANK` need to know the size of window partition. But in stream mode, we can't know the size as the data is unbounded.

Comment (nyingping) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala:112]: Yeah, I mean the exception cue is 'batch mode'.

Comment (luoyuxia) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala:112]: Thanks for pointing it.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**